File: fabsoft-backend/src/main/java/br/univille/fabsoft_backend/domain/entity/Customer.java
package br.univille.fabsoft_backend.domain.entity;

import java.util.List;

import br.univille.fabsoft_backend.domain.enums.ExerciseDays;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    private Integer age;
    private Integer weight;
    private Integer height;
    private String gender;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = true)  // Explicitly tells JPA this column can be null
    private ExerciseDays exerciseDays;
    
    @OneToMany(mappedBy = "customer")
    private List<ExecutedExercise> executedExercises;
}

File: fabsoft-frontend/src/app/model/customer.ts
export class CustomerModel {
  id!: number;
  name!: string;
  age!: number;
  weight!: number;
  height!: number;
  exerciseDays!: ExerciseDays;
  gender!: 'Male' | 'Female';
}

export enum ExerciseDays {
  THREE_DAYS = 'THREE_DAYS',
  FOUR_DAYS = 'FOUR_DAYS',
  FIVE_DAYS = 'FIVE_DAYS',
  MORE_THAN_FIVE = 'MORE_THAN_FIVE'
}

